### PR TITLE
Add GitHub Actions build.yml workflow

### LINKED ISSUE #1: Please consider adding a GitHub Actions build workflow
I understand that the **GitLab** CI/CD pipeline is no longer used since the project has been moved here. So it might be a great idea to replace it with one or more **GitHub Actions** workflow(s).

### PR DESCRIPTION
**Pull request by AdrianDsg**
fix #1

### CODE REVIEW

**Comment by fmoessbauer** [comment]
Thanks for this contribution. Before we can merge this, please adapt it to the requirements stated in the CONTRIBUTING.md guide. Mainly:

- sign-off your commit
- add the license header

Further, why do you limit the CI to the main branch? It would be good to always run it (after approval by a maintainer). Otherwise we will only discover issues once already merged.

**Review by nejch — CHANGES_REQUESTED** [review]
I guess it makes sense to also remove `.gitlab-ci.yml` in this PR :) 

(not a maintainer here, just subscribed)

**Comment by jan-kiszka** [comment]
FWIW, I've requested to add the [DCO app](https://github.com/apps/dco) to this repo. We are using it elsewhere already.

**Comment by jan-kiszka** [comment]
> FWIW, I've requested to add the [DCO app](https://github.com/apps/dco) to this repo. We are using it elsewhere already.

And it works :laughing:

**Comment by jan-kiszka** [comment]
Signing != signed-off-by tag. Your first commit has something like that, but with a pseudo email. `git commit --amend -s`, making sure that your local git configuration carries the right email address.

File: .github/workflows/build.yml
Comment (jan-kiszka) [inline .github/workflows/build.yml:64]: Extra black line at EOF.

**Comment by bufferoverflow** [comment]
I will create a new MR based on this with my email.

**Pull request closed without merging**
